### DISCUSSION TITLE
Node Callbacks not working in Angular

### DISCUSSION THREAD

**Discussion by niclasschult** [cytoscape/cytoscape.js#3190] [General]
Hi everyone,
a while ago I used Cytoscape JS a lot for an interactive data exploration tool and it worked great for me. Now I am trying to use it inside a new Angular app and while graph rendering works fine, I have problems with user interactions. In particular, the node and edge callbacks do not work at all (using cy.on(...)) .  Even very simple callbacks like

```js
 this.cy.on('click', 'node', function (evt) {
      console.log('clicked ');
    });

```

just do not do anything. I tried to add a breakpoint inside of the callback function and the function is never triggered. The click events seem to be propagated to the nodes though, since selection of nodes works without issues.  I also tried different variants, eg, using addListener() instead of "on()", using cy.nodes().on(...) and similar, but without success.  Were there major changes to the way node callbacks work in the last few years? Or are there known problems with cytoscape in Angular?  I am sorry, I really do not understand what I am doing wrong...
    
Any help would be greatly appreciated!    Thanks very much in advance!

ps  cy.ready() callbacks, for example, work fine. I only have problems with callbacks on nodes and edges (tap, mouseover, cxttap etc)

**Comment by maxkfranz** [comment]
Sounds like an Angular bug or an app bug. Double check your listener registration is actually called and not errantly cleaned up in your app. If needed after that, contact the Angular team, as then they may have a bug.

**Comment by niclasschult** [comment]
Yes, sorry, it was a bug in my app. The graph was recreated after I added the node event callbacks... sorry for wasting your time...
